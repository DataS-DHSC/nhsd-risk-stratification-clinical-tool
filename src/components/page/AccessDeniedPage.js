import Head from 'next/head';
import { Row, Col, Label } from 'nhsuk-react-components';
import {
  ROUTE_PATIENT_GUIDANCE,
  ROUTE_LOGIN,
  ROUTE_ACCESS_INFO,
} from '../../constants/routes';
import { SITE_BRANDING_TITLE } from '../../constants/site-branding';

const AccessDeniedPage = () => (
  <>
    <Head>
      <title>{SITE_BRANDING_TITLE} - Access denied</title>
      <meta
        name="description"
        content={`${SITE_BRANDING_TITLE} access denied`}
      />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>
    <Row>
      <Col width="two-thirds">
        <Label isPageHeading>
          You&#39;ve not yet been given access to the COVID-19 Clinical Risk{' '}
          Assessment Tool
        </Label>
        <p>
          If you&#39;re a clinician and want to use the COVID-19 Clinical Risk
          Assessment Tool,{' '}
          <a id="login-link" href={ROUTE_LOGIN}>
            {' '}
            generate a link to sign in.
          </a>
        </p>
        <p>
          As a clinician, you can{' '}
          <a
            id="clinician-info-link"
            href={ROUTE_ACCESS_INFO}
            target="_blank"
            rel="noreferrer"
          >
            find out how to get access to the tool (opens in new window).
          </a>
        </p>
        <p>
          If you&#39;re not a clinician and you have a question about the tool,
          you should read the{' '}
          <a
            id="patient-guidance-link"
            href={ROUTE_PATIENT_GUIDANCE}
            target="_blank"
            rel="noreferrer"
          >
            Information for Patients (opens in new window)
          </a>{' '}
          or email risk.strat.spl@nhs.net
        </p>
      </Col>
    </Row>
  </>
);

export default AccessDeniedPage;
