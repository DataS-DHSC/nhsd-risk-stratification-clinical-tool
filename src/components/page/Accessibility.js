import React from 'react';

const Accessibility = () => (
  <>
    <div className="nhsuk-grid-row">
      <div className="nhsuk-grid-column-two-thirds">
        <h1>Redacted for open source publication reasons</h1>
      </div>
    </div>
  </>
);

export default Accessibility;
