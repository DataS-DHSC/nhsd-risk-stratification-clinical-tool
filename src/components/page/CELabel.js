import React from 'react';

const CELabel = () => (
  <>
    <div className="nhsuk-grid-row">
      <div className="nhsuk-grid-column-two-thirds">
        <h1>Redacted for open source publication reasons</h1>
      </div>
    </div>
  </>
);

export default CELabel;
