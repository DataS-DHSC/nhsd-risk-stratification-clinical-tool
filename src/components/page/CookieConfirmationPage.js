import React from 'react';
import PageReviewDates from '../PageReviewDates';

export function CookieConfirmationPage() {
  return (
    <>
      <div className="nhsuk-grid-row">
        <div className="nhsuk-grid-column-two-thirds">
          <h1>Your cookie settings have been saved.</h1>
        </div>
      </div>
      <article>
        <div className="nhsuk-grid-row">
          <div className="nhsuk-grid-column-two-thirds">
            <section>
              <div className="block-richtext">
                <p>We&apos;ll save your settings for a year.</p>
                <p>
                  We&apos;ll ask you if you&apos;re still OK with us using
                  cookies when either:
                </p>
                <ul>
                  <li>
                    it&apos;s been a year since you last saved your settings we
                  </li>
                  <li>add any new cookies or change the cookies we use</li>
                </ul>
                <p>
                  You can also change your settings at any time using{' '}
                  <a href="/page/cookies/policy">our cookies page.</a>
                </p>
              </div>
            </section>
          </div>
        </div>
      </article>
      <PageReviewDates page="static" />
    </>
  );
}
