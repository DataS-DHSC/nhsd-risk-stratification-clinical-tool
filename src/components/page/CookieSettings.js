/* eslint-disable jsx-a11y/label-has-associated-control */
import { addMonths } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react';
import cookie from 'react-cookies';
import { useCookiesContext } from '../../context/cookies-context';
import {
  boolToAnalyticsInputvalue,
  analyticsInputValueToBool,
} from '../../utils/cookieSettingsUtils';
import { ROUTE_COOKIES_CONFIRMATION } from '../../constants/routes';

function CookieSettings() {
  const router = useRouter();

  const { optIn, setOptIn } = useCookiesContext();

  const [analyticsCookiesInput, setAnalyticsCookiesInput] = useState(
    boolToAnalyticsInputvalue(optIn)
  );

  function handleSubmit(evt) {
    evt.preventDefault();
    const defaultCookieSettings = {
      path: '/',
      sameSite: 'Strict',
    };
    cookie.save(
      'opt-in',
      `${analyticsInputValueToBool(analyticsCookiesInput)}`,
      {
        ...defaultCookieSettings,
        expires: addMonths(new Date(), 12),
      }
    );
    setOptIn(analyticsInputValueToBool(analyticsCookiesInput));
    router.push(ROUTE_COOKIES_CONFIRMATION);
  }

  return (
    <body>
      <div className="nhsuk-grid-row">
        <div className="nhsuk-grid-column-two-thirds">
          <h1 className="nhsuk-heading-l">Change your cookie settings</h1>
          <noscript>
            <p className="nhsuk-body">
              We cannot change your cookie settings at the moment because
              JavaScript is not running in your browser. To fix this, try:
            </p>
            <ul className="nhsuk-list nhsuk-list--bullet">
              <li>turning on JavaScript in your browser settings</li>
              <li>reloading this page</li>
            </ul>
          </noscript>

          <form onSubmit={handleSubmit} method="post" noValidate>
            <div className="nhsuk-form-group">
              <p className="nhsuk-body">
                Let us know which cookies we can use. We&apos;ll use a cookie to
                save your settings.
              </p>
              <fieldset className="nhsuk-fieldset">
                <legend className="nhsuk-fieldset__legend nhsuk-fieldset__legend--s">
                  Cookies that measure website use (analytics cookies)
                </legend>
                <p>
                  These cookies store information about how you use our website,
                  such as the links you click on. We use them to help make our
                  website better.
                </p>
                <div className="nhsuk-radios">
                  <div className="nhsuk-radios__item">
                    <input
                      className="nhsuk-radios__input"
                      id="analytics-cookies-yes"
                      name="analytics-cookies"
                      type="radio"
                      value="yes"
                      onChange={(evt) => {
                        setAnalyticsCookiesInput(evt.target.value);
                      }}
                      checked={analyticsCookiesInput === 'yes'}
                    />
                    <label
                      className="nhsuk-label nhsuk-radios__label"
                      htmlFor="analytics-cookies-yes"
                    >
                      Yes
                    </label>
                  </div>
                  <div className="nhsuk-radios__item">
                    <input
                      className="nhsuk-radios__input"
                      id="analytics-cookies-no"
                      name="analytics-cookies"
                      type="radio"
                      value="no"
                      onChange={(evt) => {
                        setAnalyticsCookiesInput(evt.target.value);
                      }}
                      checked={analyticsCookiesInput === 'no'}
                    />
                    <label
                      htmlFor="analytics-cookies-no"
                      className="nhsuk-label nhsuk-radios__label"
                    >
                      No
                    </label>
                  </div>
                </div>
              </fieldset>
            </div>

            <button
              type="submit"
              className="nhsuk-button"
              data-module="nhsuk-button"
              id="save-cookie-settings"
            >
              Save cookie settings
            </button>
          </form>
        </div>
      </div>
    </body>
  );
}

export default CookieSettings;
