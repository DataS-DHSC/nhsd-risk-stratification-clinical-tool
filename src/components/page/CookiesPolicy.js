import React from 'react';
import PageReviewDates from '../PageReviewDates';
import SiteBrandingTitle from '../SiteBrandingTitle';
import { ROUTE_COOKIES_SETTINGS } from '../../constants/routes';

const CookiesPolicy = () => (
  <>
    <div className="nhsuk-grid-row">
      <div className="nhsuk-grid-column-two-thirds">
        <h1>
          Cookies for the <SiteBrandingTitle isDefaultName />
        </h1>
      </div>
    </div>
    <article>
      <div className="nhsuk-grid-row">
        <div className="nhsuk-grid-column-two-thirds">
          <section>
            <div className="block-richtext">
              <p>
                Cookies are small files saved on your phone, tablet or computer
                when you visit a website.
              </p>
              <p>
                They store information about how you use the website. Some of
                them can be turned on and off.
              </p>
              <h2>Essential cookies</h2>
              <p>
                Essential cookies keep your information secure while you use
                this service. We do not need to ask your permission to use them.
              </p>
              <table>
                <thead>
                  <tr>
                    <th scope="col">Name</th>
                    <th scope="col">Purpose</th>
                    <th scope="col">Expires</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>__Host-csrfToken</td>
                    <td>
                      Helps keep the site secure by preventing cross-site
                      request forgery (CSRF) attacks
                    </td>
                    <td>
                      This cookie is a session cookie and it will expire when
                      the browser window is closed.
                    </td>
                  </tr>
                  <tr>
                    <td>correlationId</td>
                    <td>
                      This is used to determine one user of the service from
                      another. It is also used when we need to investigate any
                      issues you may raise with the service.
                    </td>
                    <td>
                      This cookie is a session cookie and it will expire when
                      the browser window is closed.
                    </td>
                  </tr>
                  <tr>
                    <td>access_token</td>
                    <td>
                      This is used as part of our authorization process.
                      It&apos;s sent with your requests to our servers so we can
                      verify your identity.
                    </td>
                    <td>24 hours</td>
                  </tr>
                  <tr>
                    <td>refresh_token</td>
                    <td>
                      This is used in the process of getting a new access_token
                      when your current access_token expires.
                    </td>
                    <td>6 months</td>
                  </tr>
                </tbody>
              </table>
              <h2>Functional Cookies</h2>
              <p>
                The service will still work without functional cookies, however
                you may not be able to take advantage of certain functionality,
                such as the ability to remember your settings preferences
                between visits.
              </p>
              <table>
                <thead>
                  <tr>
                    <th scope="col">Name</th>
                    <th scope="col">Purpose</th>
                    <th scope="col">Expires</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>opt-in</td>
                    <td>Saves your cookie consent settings</td>
                    <td>1 year</td>
                  </tr>
                </tbody>
              </table>
              <h2>Analytics Cookies (Optional)</h2>
              <p>
                With your permission, we use Adobe Analytics to gather
                information about how you use this service. This information
                helps us improve our service.
              </p>
              <p>
                Adobe is not allowed to share our analytics data with anyone.
              </p>
              <p>
                Adobe stores anonymised information about:
                <br />
                <ul>
                  <li>How long you spend on the service.</li>
                  <li>Any errors you see while using the service.</li>
                </ul>
              </p>
              <table>
                <thead>
                  <tr>
                    <th scope="col">Name</th>
                    <th scope="col">Purpose</th>
                    <th scope="col">Expires</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>s_sq</td>
                    <td>
                      This cookie is used by Adobe Analytics. This cookie is set
                      and read by the JavaScript code when the ClickMap
                      functionality or the Activity Map functionality are
                      enabled; it contains information about the previous link
                      that was clicked on by the user
                    </td>
                    <td>
                      This cookie is a session cookie and it will expire when
                      the browser window is closed.
                    </td>
                  </tr>
                  <tr>
                    <td>s_cc</td>
                    <td>
                      This cookie is used by Adobe Analytics. This cookie is set
                      and read by the JavaScript code to determine if cookies
                      are enabled (simply set to “True”)
                    </td>
                    <td>
                      This cookie is a session cookie and it will expire when
                      the browser window is closed.
                    </td>
                  </tr>
                  <tr>
                    <td>s_getNewRepeat</td>
                    <td>
                      This cookie is used by Adobe Analytics. It determines if a
                      visitor is new to site or not.
                    </td>
                    <td>1 month</td>
                  </tr>
                  <tr>
                    <td>s_ppn</td>
                    <td>
                      This cookie is used by Adobe Analytics. It stores data on
                      device recognition.
                    </td>
                    <td>1 day</td>
                  </tr>
                  <tr>
                    <td>s_fid</td>
                    <td>
                      This cookie is used by Adobe Analytics. This cookie is
                      used to identify a unique visitor if the standard s_vi
                      cookie is unavailable due to third-party cookie
                      restrictions. Not used for implementations that use
                      first-party cookies.
                    </td>
                    <td>2 years</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div lassName="block-richtext">
              <h2>Change your cookie settings</h2>
              <p>
                Some cookies, like those used to measure how you use our
                website, are not needed for our website to work.
              </p>
              <p>
                These cookies can help us make our website better, but
                we&apos;ll only use them if you say it&apos;s OK.{' '}
              </p>
              <div className="nhsuk-action-link">
                <a
                  className="nhsuk-action-link__link"
                  href={ROUTE_COOKIES_SETTINGS}
                  id="choose-cookies-link"
                >
                  <svg
                    className="nhsuk-icon nhsuk-icon__arrow-right-circle"
                    xmlns="http://www.w3.org/2000/svg"
                    viewBox="0 0 24 24"
                    aria-hidden="true"
                  >
                    <path d="M0 0h24v24H0z" fill="none" />
                    <path d="M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z" />
                  </svg>
                  <span className="nhsuk-action-link__text">
                    Choose which cookies we use
                  </span>
                </a>
              </div>
            </div>
          </section>
        </div>
      </div>
    </article>
    <PageReviewDates page="static" />
  </>
);

export default CookiesPolicy;
