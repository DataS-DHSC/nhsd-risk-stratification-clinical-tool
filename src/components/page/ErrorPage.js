import Head from 'next/head';
import PropTypes from 'prop-types';
import cookie from 'react-cookies';
import { ROUTE_IP_ENDPOINT, ROUTE_SUPPORT_FORM } from '../../constants/routes';
import { SITE_BRANDING_TITLE } from '../../constants/site-branding';

const errors = [
  {
    code: 400,
    title: 'Something went wrong',
  },
  {
    code: 404,
    title: 'Page not found',
    message:
      'If you entered a web address, make sure you entered it correctly.\n\nIf you pasted the web address, make sure you copied the entire address. ',
  },
  {
    code: 503,
    title: 'Service unavailable',
    message:
      'Sorry, but this service is currently unavailable whilst a new version is being released. It will be available again soon. Please try again later.',
  },
];

const ErrorPage = ({ statusCode, startAgainLink }) => {
  let error = errors.find((e) => e.code === statusCode);
  if (!error) error = errors.find((e) => e.code === 400);

  const correlationId = cookie.load('correlationId') || 'ead1c4fe';

  return (
    <>
      <Head>
        <title>{SITE_BRANDING_TITLE} - Error</title>
        <meta name="description" content={`${SITE_BRANDING_TITLE} error`} />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      </Head>
      <div className="nhsuk-grid-row">
        <div className="nhsuk-grid-column-full">
          <h1 id="error-heading">{error.title}</h1>
        </div>
        <div className="nhsuk-grid-column-two-thirds">
          {statusCode === 400 || statusCode === 403 ? (
            <p id="temporary-problem-message">
              Sorry, an error has occurred with the service. This is usually a
              temporary problem, but if it continues you should let our Helpdesk
              know.
              <br />
              <br />
              If you need to contact the Helpdesk:
              <br />
              <br />
              You may first need to{' '}
              <a
                href={ROUTE_IP_ENDPOINT}
                rel="noreferrer nofollow noopener"
                target="_blank"
              >
                find your IP address (opens in new window)
              </a>
              {correlationId &&
                ` and enter that as well as your Correlation ID: ${correlationId} `}
              in ‘Your message for us’ when you{' '}
              <a
                rel="noreferrer nofollow noopener"
                target="_blank"
                href={encodeURI(`${ROUTE_SUPPORT_FORM}`)}
                id="helpdesk-link"
              >
                complete the helpdesk contact form (opens in new window)
              </a>
            </p>
          ) : (
            <p style={{ whiteSpace: 'pre-wrap' }}>{error.message}</p>
          )}

          {startAgainLink && (
            <a
              href={startAgainLink}
              className="nhsuk-u-margin-top-5 nhsuk-u-font-size-19"
              id="new-assessment-link"
            >
              Start a new COVID-19 risk assessment
            </a>
          )}
        </div>
      </div>
    </>
  );
};

ErrorPage.propTypes = {
  statusCode: PropTypes.number.isRequired,
  startAgainLink: PropTypes.string,
};

ErrorPage.defaultProps = {
  startAgainLink: '',
};

export default ErrorPage;
