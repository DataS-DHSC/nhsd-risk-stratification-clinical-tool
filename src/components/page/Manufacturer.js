import React from 'react';

export const Manufacturer = () => (
  <div className="nhsuk-summary-list__row">
    <dt className="nhsuk-summary-list__key legal-list__title">
      <img src="/icons/ce-manufacturer.svg" alt="Manufacturer" />
    </dt>
    <dt className="nhsuk-summary-list__key">Manufacturer</dt>
  </div>
);
