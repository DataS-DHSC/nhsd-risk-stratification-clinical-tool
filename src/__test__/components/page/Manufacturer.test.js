import React from 'react';
import { render } from '@testing-library/react';
import { Manufacturer } from '../../../components/page/Manufacturer';

const setup = () => render(<Manufacturer />);

describe('Manufacturer', () => {
  it('should match snapshot', () => {
    const container = setup();

    expect(container.asFragment()).toMatchSnapshot();
  });
});
