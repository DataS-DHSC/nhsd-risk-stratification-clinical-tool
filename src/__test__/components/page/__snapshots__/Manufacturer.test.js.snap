// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manufacturer should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-summary-list__row"
  >
    <dt
      class="nhsuk-summary-list__key legal-list__title"
    >
      <img
        alt="Manufacturer"
        src="/icons/ce-manufacturer.svg"
      />
    </dt>
    <dt
      class="nhsuk-summary-list__key"
    >
      Manufacturer
    </dt>
  </div>
</DocumentFragment>
`;
